src/back/space/src/api/invitation.cpp

```diff
@@ -0,0 +1,163 @@
+#include "../service/service.hpp"
+#include "../../../shared/headers.hpp"
+#include "../../../shared/errors.hpp"
+#include "../../../shared/paging.hpp"
+#include "../../../shared/parse/request.hpp"
+#include "../model/role.hpp"
+#include "../model/invitation_serialize.hpp"
+
+namespace svetit::space::handlers {
+
+Invitation::Invitation(
+	const components::ComponentConfig& conf,
+	const components::ComponentContext& ctx)
+	: server::handlers::HttpHandlerJsonBase{conf, ctx}
+	, _s{ctx.FindComponent<Service>()}
+{}
+
+formats::json::Value Invitation::HandleRequestJsonThrow(
+	const server::http::HttpRequest& req,
+	const formats::json::Value& body,
+	server::request::RequestContext&) const
+{
+	formats::json::ValueBuilder res;
+
+	const auto userId = req.GetHeader(headers::kUserId);
+	if (userId.empty())
+		throw errors::Unauthorized{};
+
+	try {
+		switch (req.GetMethod()) {
+		case server::http::HttpMethod::kGet:
+			return GetList(req, res, userId);
+		case server::http::HttpMethod::kPost:
+			return Post(req, body, res, userId);
+		case server::http::HttpMethod::kPut:
+			return ChangeRole(req, body, res, userId);
+		case server::http::HttpMethod::kPatch:
+			return Join(req, res, userId);
+		case server::http::HttpMethod::kDelete:
+			return Delete(req, res, userId);
+		default:
+			throw std::runtime_error("Unsupported");
+			break;
+		}
+	} catch(const errors::Unauthorized& e) {
+		req.SetResponseStatus(server::http::HttpStatus::kUnauthorized);
+		res["err"] = e.what();
+		return res.ExtractValue();
+	} catch(const errors::BadRequest& e) {
+		req.SetResponseStatus(server::http::HttpStatus::kBadRequest);
+		res["err"] = e.what();
+		return res.ExtractValue();
+	} catch(const errors::NotFound& e) {
+		req.SetResponseStatus(server::http::HttpStatus::kNotFound);
+		res["err"] = e.what();
+		return res.ExtractValue();
+	} catch(const errors::NotModified& e) {
+		req.SetResponseStatus(server::http::HttpStatus::kNotModified);
+		res["err"] = e.what();
+		return res.ExtractValue();
+	} catch(const std::exception& e) {
+		LOG_WARNING() << "Fail to process invitation handle with method: "
+			<< req.GetMethodStr() << " err: " << e.what();
+		res["err"] = "Fail to process invitation";
+		req.SetResponseStatus(server::http::HttpStatus::kInternalServerError);
+	}
+	return res.ExtractValue();
+}
+
+formats::json::Value Invitation::GetList(
+	const server::http::HttpRequest& req,
+	formats::json::ValueBuilder& res,
+	const std::string& userId) const
+{
+	auto paging = parsePaging(req);
+	if (_s.IsListLimit(paging.limit))
+		throw errors::BadRequest("Too big limit param");
+
+	PagingResult<model::SpaceInvitation> list;
+	std::string spaceId;
+	if (req.HasArg("spaceId")) {
+		spaceId = req.GetArg("spaceId");
+		if (spaceId.empty())
+			throw errors::BadRequest("SpaceId param shouldn't be empty");
+		list = _s.GetInvitationListBySpaceForSpaceDetail(utils::BoostUuidFromString(spaceId), paging.start, paging.limit, userId);
+	} else {
```

Comment:
Посмотри пожалуйста везде где парсим uuid нужно использовать нашу функцию parseUUID

```suggestion
	if (req.HasArg("spaceId")) {
		const auto spaceId = parseUUID(req, "spaceId");
		list = _s.GetInvitationListBySpaceForSpaceDetail(spaceId, paging.start, paging.limit, userId);
	} else {
```

Reply:
Готово.